bigquery_etl/cli/query.py

```diff
@@ -1479,64 +1479,77 @@ def deploy(
+                    f"run `./bqetl query schema update "
+                    f"{dataset_name}.{table_name}`",
+                    err=True,
                 )
-                if not existing_schema.equal(query_schema):
-                    click.echo(
-                        f"Query {query_file_path} does not match "
-                        f"schema in {existing_schema_path}. "
-                        f"To update the local schema file, "
-                        f"run `./bqetl query schema update "
-                        f"{dataset_name}.{table_name}`",
-                        err=True,
-                    )
-                    sys.exit(1)
+                sys.exit(1)  # TODO: Should this be a continue?
 
-            tmp_schema_file = NamedTemporaryFile()
+        with NamedTemporaryFile(suffix=".json") as tmp_schema_file:
             existing_schema.to_json_file(Path(tmp_schema_file.name))
             bigquery_schema = client.schema_from_json(tmp_schema_file.name)
 
-            try:
-                # destination table already exists, update schema
-                table = client.get_table(full_table_id)
-                table.schema = bigquery_schema
-                client.update_table(table, ["schema"])
-                click.echo(f"Schema updated for {full_table_id}")
-            except NotFound:
-                # no destination table, create new table based on schema and metadata
-                metadata = Metadata.of_query_file(query_file_path)
-                new_table = bigquery.Table(full_table_id, schema=bigquery_schema)
-
-                if metadata.bigquery and metadata.bigquery.time_partitioning:
-                    new_table.time_partitioning = bigquery.TimePartitioning(
-                        metadata.bigquery.time_partitioning.type.bigquery_type,
-                        field=metadata.bigquery.time_partitioning.field,
-                        require_partition_filter=(
-                            metadata.bigquery.time_partitioning.require_partition_filter
-                        ),
-                        expiration_ms=metadata.bigquery.time_partitioning.expiration_ms,
-                    )
+        try:
+            table = client.get_table(full_table_id)
+        except NotFound:
+            table = bigquery.Table(full_table_id)
 
-                if metadata.bigquery and metadata.bigquery.clustering:
-                    new_table.clustering_fields = metadata.bigquery.clustering.fields
+        table.schema = bigquery_schema
 
-                client.create_table(new_table)
-                click.echo(f"Destination table {full_table_id} created.")
+        if table.created is None:
+            _attach_metadata(query_file_path, table)
+            client.create_table(table)
+            click.echo(f"Destination table {full_table_id} created.")
         else:
-            click.echo(f"No schema file found for {query_file}")
+            client.update_table(table, ["schema"])
```

Comment:
We could `_attach_metadata` on every schema deploy here actually. This way changes in `metadata.yaml` would be reflected in BigQuery on `bqetl query schema deploy`. I'm hesitating because this actually seems like something that should be done on every post-merge like the way we do view deploys.

I know we also have this: https://github.com/mozilla/bigquery-etl/blob/a190e1826498da6c898cf2910af59f7a2e8f64f2/bigquery_etl/metadata/publish_metadata.py#L32 but I'm not sure when/where that gets used?